### ISSUE TITLE
Error while syncing tags

### ISSUE THREAD

**Issue by jhf2442** [baumgarr/nixnote2#149]
Hi,

running latest 0d573b759fd5ff1f7be079b32eba8da12f186c96 (==master) on arch linux (AUR)

when running sync, one tag brings an error : 
DEBUG 2016-01-18 00:03:13.584 ( ../nixnote2/threads/syncrunner.cpp @ 51 ) Starting SyncRunner.synchronize() 
DEBUG 2016-01-18 00:03:13.584 ( ../nixnote2/sql/databaseconnection.cpp @ 37 ) SQL drivers available:  ("QSQLITE", "QSQLITE3", "QMYSQL3", "QMYSQL", "QODBC3", "QODBC", "QPSQL7", "QPSQL") 
DEBUG 2016-01-18 00:03:13.674 ( ../nixnote2/communication/communicationmanager.cpp @ 120 ) Inside CommunicationManager::initNoteStore() 
DEBUG 2016-01-18 00:03:13.674 ( ../nixnote2/communication/communicationmanager.cpp @ 150 ) Inside CommunicationManager::getUserInfo 
DEBUG 2016-01-18 00:03:14.923 ( ../nixnote2/communication/communicationmanager.cpp @ 150 ) Inside CommunicationManager::getUserInfo 
DEBUG 2016-01-18 00:03:15.786 ( ../nixnote2/threads/syncrunner.cpp @ 133 ) --->>>  Current Chunk High Sequence Number:  1951 
DEBUG 2016-01-18 00:03:15.786 ( ../nixnote2/threads/syncrunner.cpp @ 134 ) --->>>  Last User High Sequence Number:  1951 
DEBUG 2016-01-18 00:03:15.786 ( ../nixnote2/communication/communicationmanager.cpp @ 150 ) Inside CommunicationManager::getUserInfo 
 INFO 2016-01-18 00:03:15.787 ( ../nixnote2/nixnote.cpp @ 1791 ) "Beginning Sync" 
ERROR 2016-01-18 00:03:16.661 ( ../nixnote2/communication/communicationmanager.cpp @ 399 ) EDAMUserException: 10 

ERROR 2016-01-18 00:03:16.661 ( ../nixnote2/communication/communicationmanager.cpp @ 400 )  Tag name:  "Crypto1" 

DEBUG 2016-01-18 00:03:16.661 ( ../nixnote2/utilities/debugtool.cpp @ 33 ) *** Dumping Tag *** 
DEBUG 2016-01-18 00:03:16.661 ( ../nixnote2/utilities/debugtool.cpp @ 215 ) "guid" : "730efa7d-f0ff-4a45-9612-eb4168ae7490" : 
DEBUG 2016-01-18 00:03:16.661 ( ../nixnote2/utilities/debugtool.cpp @ 215 ) "name" : "Crypto1" : 
DEBUG 2016-01-18 00:03:16.661 ( ../nixnote2/utilities/debugtool.cpp @ 217 ) "parent"  is not set. 
DEBUG 2016-01-18 00:03:16.662 ( ../nixnote2/utilities/debugtool.cpp @ 245 ) "USN" : 0 : 
DEBUG 2016-01-18 00:03:16.662 ( ../nixnote2/utilities/debugtool.cpp @ 38 ) *** Tag Dump complete *** 
 INFO 2016-01-18 00:03:16.662 ( ../nixnote2/nixnote.cpp @ 1791 ) "Communication Error - Data conflict : :EDAMUserException: DATA_CONFLICT parameter=Tag.name" 
 INFO 2016-01-18 00:03:16.663 ( ../nixnote2/nixnote.cpp @ 1791 ) "Communication Error - Data conflict : :EDAMUserException: DATA_CONFLICT parameter=Tag.name" 
DEBUG 2016-01-18 00:03:16.663 ( ../nixnote2/gui/ntableview.cpp @ 451 ) Valid LIDs retrieved.  Refreshing selection 
DEBUG 2016-01-18 00:03:16.674 ( ../nixnote2/nixnote.cpp @ 1475 ) starting NixNote.updateSelectionCriteria() 

Name of the tag is Crypto1 because I've read in another issue that it might help to rename the tag locally, but this didn't help

Stange thing is that sometimes it blocks on another tag - maybe the order of the tags in the sync list is not always the same ?

Thanks in advance

**Comment by baumgarr** [comment]
Look at your Evernote account using the web client.  Is there a tag called Crypto1 already in your account?

Is this tag being created in a shared notebook that somebody else owns?  Is it a notebook that you own but is shared with somebody else?  Was the tag created directly by left-clicking on the tag list & choosing "Add" or was it created by typing a new tag name in the note field?

**Comment by baumgarr** [comment]
I think I have this fixed.  It looks like a duplicate of Data_conflict parameter=Tag.name #147.

It should be fixed in the next release, but you may have to do some work (depending on the note).  Look at at issue for further details.

**Comment by jhf2442** [comment]
Hi baumgarr,

tags were created by typing the name in the tag field of a note. I have no shared notebooks, only working with myself :-)

I've compiled yesterday's code base and ran nixnote2 afterwards. the crypto1 tag was then visible in the EN web interface, but on the other side the crypto tag did not appear with nixnote (well... there as a tag with an empty name visible, and it was associated with the articles that should have had the crypto tag), even after multiple restarts and sync. I've finally ressorted to fixing the tags manually in nixnote as there were only ~10 articles involved

Anyhow, thanks for the quick fix. Will monitor the articles and check that sync now runs OK and tags don't get corrupted anew.
